### PR TITLE
Fix issue link does not support quotes (#29484)

### LINKED ISSUE #29484: alphanumeric issue matching not working with quotes around issue
### Description

I've enabled external issue tracking with an alphanumeric issue number format in my repository. The issue is linked correctly in commit messages, except when the ID is surrounded by a quote, as in git's automatic revert message format:

```
Revert "ABC-123 - add README"
```

See this on the demo site here: https://try.gitea.io/hmoffatt/hamish-test/commits/branch/main

### Gitea Version

1.21.5

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Screenshots

Firefox_Screenshot_2024-02-29T01-12-22 862Z

### Git Version

_No response_

### Operating System

Debian

### How are you running Gitea?

Binary from your releases.

### Database

PostgreSQL

### PR DESCRIPTION
**Pull request by charles7668**
Close #29484

圖片

### CODE REVIEW

**Comment by lunny** [comment]
Please add some tests

**Comment by yp05327** [comment]
Didn't link the issue correctly.
https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

**Comment by charles7668** [comment]
> Didn't link the issue correctly. https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

Fixed

**Review by lunny — APPROVED** [review]

File: modules/references/references.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ var (
 	// issueNumericPattern matches string that references to a numeric issue, e.g. #1287
 	issueNumericPattern = regexp.MustCompile(`(?:\s|^|\(|\[|\')([#!][0-9]+)(?:\s|$|\)|\]|[:;,.?!]\s|[:;,.?!]$)`)
```
Comment (yp05327) [inline modules/references/references.go]: Just a question, should we also add `"` support here? 
Something like `"#1234"` will be failed now.
And same to others.

**Review by delvh — APPROVED** [review]

File: modules/references/references.go (hunk 1 above)
Comment (lunny) [reply modules/references/references.go]: @charles7668 I think this is the last one we should fix in this PR.

Comment (charles7668) [reply modules/references/references.go]: > @charles7668 I think this is the last one we should fix in this PR.

I fixed it

**Pull request merged**